gradle/validation/check-broken-links.gradle

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+configure(rootProject) {
+
+  task checkBrokenLinks {
+    group 'Verification'
+    description 'Check broken links in the entire documentation'
+
+    dependsOn ':lucene:checkBrokenLinks'
+    dependsOn ':solr:checkBrokenLinks'
+  }
+
+}
+configure(subprojects.findAll { it.path in [':lucene', ':solr'] }) {
+
+  task checkBrokenLinks(type: CheckBrokenLinksTask, 'dependsOn': 'documentation')
+
+  // TODO: uncomment this line after fixing all broken links.
+  // (we can't fix the cross-project links until ant build is disabled.)
+  // check.dependsOn checkBrokenLinks
+}
+
+class CheckBrokenLinksTask extends DefaultTask {
+
+  @Input
```

Comment:
Should be `@InputFile`

Reply:
I replaced it with `@InputDirectory` (since `@InputFile` does not work for me.)

Reply:
of course it's a directory, sorry!

Reply:
For reference - the difference is that Input is  just the file, InputFIle is the file and its contents (!). 
https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/InputFile.html

Reply:
This works for me and is what I had in mind:
```
 configure(subprojects.findAll { it.path in [':lucene', ':solr'] }) {
 
-  task checkBrokenLinks(type: CheckBrokenLinksTask, 'dependsOn': 'documentation')
+  task checkBrokenLinks(type: CheckBrokenLinksTask, 'dependsOn': 'documentation') {
+    docsDir = project.layout.dir provider { project.docroot }
+  }
 
   // TODO: uncomment this line after fixing all broken links.
   // (we can't fix the cross-project links until ant build is disabled.)
@@ -38,7 +39,7 @@ configure(subprojects.findAll { it.path in [':lucene', ':solr'] }) {
 class CheckBrokenLinksTask extends DefaultTask {
 
   @InputDirectory
-  final Provider<File> docsDir = project.providers.provider { project.docroot }
+  final DirectoryProperty docsDir = project.objects.directoryProperty()
 
   @InputFile
   File script = project.rootProject.file("dev-tools/scripts/checkJavadocLinks.py")
@@ -56,7 +57,7 @@ class CheckBrokenLinksTask extends DefaultTask {
         args = [
           "-B",
           script.absolutePath,
-          docsDir.get().absolutePath
+          docsDir.get().asFile
         ]
       }
     } 
```

Any version is fine though, I don't mind.

Reply:
Thank you, https://github.com/apache/lucene-solr/pull/1522/commits/778701c97120fbc2e6578a545d7719ac19d82d75 works for me.

Reply:
Thanks!